app/src/main/java/org/simple/clinic/teleconsultlog/prescription/medicines/TeleconsultMedicineItem.kt

```diff
@@ -0,0 +1,77 @@
+package org.simple.clinic.teleconsultlog.prescription.medicines
+
+import androidx.recyclerview.widget.DiffUtil
+import io.reactivex.subjects.Subject
+import kotlinx.android.synthetic.main.list_item_teleconsult_medicine.*
+import org.simple.clinic.R
+import org.simple.clinic.drugs.PrescribedDrug
+import org.simple.clinic.teleconsultlog.medicinefrequency.MedicineFrequency
+import org.simple.clinic.widgets.ItemAdapter
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.recyclerview.ViewHolderX
+import org.simple.clinic.widgets.visibleOrGone
+
+data class TeleconsultMedicineItem(
+    val prescribedDrug: PrescribedDrug,
+    val isLastItem: Boolean
+) : ItemAdapter.Item<UiEvent> {
+
+  companion object {
+
+    fun from(medicines: List<PrescribedDrug>): List<TeleconsultMedicineItem> {
+      return medicines
+          .mapIndexed { index, prescribedDrug ->
+            val isLastItem = index == medicines.lastIndex
+            TeleconsultMedicineItem(
+                prescribedDrug = prescribedDrug,
+                isLastItem = isLastItem
+            )
+          }
+    }
+  }
+
+  private val defaultMedicineDurationInDays = 30
+  private val defaultMedicineFrequency = MedicineFrequency.OD
+
+  override fun layoutResId(): Int = R.layout.list_item_teleconsult_medicine
+
+  override fun render(holder: ViewHolderX, subject: Subject<UiEvent>) {
+    val context = holder.itemView.context
+    holder.medicineNameTextView.text = context.getString(
+        R.string.list_item_teleconsult_medicines_name,
+        prescribedDrug.name,
+        prescribedDrug.dosage
+    )
+
+    val frequency = prescribedDrug.frequency ?: defaultMedicineFrequency
+    holder.medicineFrequencyButton.text = frequency.toString()
+
+    val durationInDays = prescribedDrug.durationInDays ?: defaultMedicineDurationInDays
+    holder.medicineDurationButton.text = context.resources.getQuantityString(
+        R.plurals.list_item_teleconsult_medicine_duration,
+        durationInDays,
+        durationInDays.toString()
+    )
+
+    holder.medicineFrequencyButton.setOnClickListener {
+      subject.onNext(DrugFrequencyClicked(prescribedDrug))
+    }
+
+    holder.medicineDurationButton.setOnClickListener {
+      subject.onNext(DrugDurationClicked(prescribedDrug))
+    }
+
+    val showDivider = isLastItem.not()
+    holder.medicinesDivider.visibleOrGone(showDivider)
```

Comment:
Yes, I completely forgot about that 😅

Reply:
Clearly you've been on the project for a good bit of time now since you're at that place where you've forgotten about stuff that you've done in the past 😂

Reply:
😅